### ISSUE TITLE
Holding KC_RSFT does not work when Auto Shift is active (it gets released before next keypress)

### ISSUE THREAD

**Issue by teddympq** [qmk/qmk_firmware#24341]
### Describe the Bug

When AutoShift is activated, holding KC_LSFT sends the corresponding shifted symbol or letter.
However, for some reason, KC_RSFT releases before pressing another key.

I am not sure whether this is the intended behavior, but I decided to tinker [quantum/process_keycode/process_autocorrect.c](url) a little bit anyway.
I am by no means an expert, this is just a suggestion or question.

In line 161, I added MOD_BIT(KC_RSFT):
Before:

> https://github.com/qmk/qmk_firmware/blob/fd65bd5ae014565d333195285b154d9eeca9294b/quantum/process_keycode/process_auto_shift.c#L161
> 
After:
`        ) & (~(MOD_BIT(KC_LSFT) | MOD_BIT(KC_RSFT)))`

In line 217, I added MOD_BIT(KC_RSFT)

> https://github.com/qmk/qmk_firmware/blob/fd65bd5ae014565d333195285b154d9eeca9294b/quantum/process_keycode/process_auto_shift.c#L217
> 
After:
`    autoshift_flags.lastshifted = (get_mods() | get_oneshot_mods()) & (MOD_BIT(KC_LSFT) | MOD_BIT(KC_RSFT));`
> 

I tested it and it seems to be working, but I am not sure if it may cause further problems, so I decided to check with you.

### Operating System

Windows

### Is AutoHotKey / Karabiner installed

- [ ] AutoHotKey (Windows)
- [ ] Karabiner (macOS)

**Comment by IsaacElenbaas** [comment]
Yes, this was missed and your suggestions are correct. 218 and 220 should also be changed.

This issue kind of shows how effective Auto Shift is haha - it has been present for three years. There aren't *that* many people who use Auto Shift, but I guess **none** of them use the regular shift keys in parallel for any reason.
